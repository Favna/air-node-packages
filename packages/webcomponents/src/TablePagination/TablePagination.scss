@import '../styles/vars';

.tablePagination {
  @extend %beautifulCenter;
  align-items: center;
  background-color: $skyBlue20;
  display: flex;
  float: left;
  font-size: $fontSizeDefault;
  height: 3rem;
  width: 100%;

  &Mobile {
    height: 2.5rem;
  }
}

.selectIcon {
  left: 1.4rem;
}

.menuItem {
  font-size: $fontSizeDefault;
}
