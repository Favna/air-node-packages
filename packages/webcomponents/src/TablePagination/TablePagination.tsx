import MUITablePagination, { TablePaginationTypeMap, TablePaginationBaseProps } from '@material-ui/core/TablePagination';
import useMediaQuery from '@material-ui/core/useMediaQuery';
import classnames from 'classnames';
import React, { FC, memo } from 'react';
import css from './TablePagination.scss';
import TablePaginationActions from '../TablePaginationActions/TablePaginationActions';
import { label, dataQa, customCss } from '../constants';

export type MutatedMUITablePaginationProps = TablePaginationTypeMap<{}, React.ComponentType<TablePaginationBaseProps>>['props'];

export interface TablePaginationProps extends MutatedMUITablePaginationProps {
  /** The label in the displayed rows per page, for example "5 rows per page" */
  labelRowsPerPage: label;
  /** The label in the displayed pages, for example "of" in "page 1 of 10" */
  labelPaginationOf: label;
  /** Data-qa tag to apply to the search bar and input element */
  'data-qa'?: dataQa;
  /** Custom CSS classes to pass to the button */
  customClasses?: customCss;
}

const TablePagination: FC<TablePaginationProps> = props => {
  const isOnMobile = useMediaQuery('(max-width: 1024px)');

  return (
    <MUITablePagination
      component='span'
      count={props.count} page={props.page}
      labelRowsPerPage={props.labelRowsPerPage}
      labelDisplayedRows={({ from, to, count }) => `${from <= 9 ? `0${from}` : from}-${to} ${props.labelPaginationOf} ${count}`}
      onChangePage={props.onChangePage} onChangeRowsPerPage={props.onChangeRowsPerPage}
      className={classnames(css.tablePagination, { [css.tablePaginationMobile]: isOnMobile }, props.customClasses)}
      classes={{ root: css.text, selectIcon: css.selectIcon, menuItem: css.text }}
      rowsPerPage={props.rowsPerPage} rowsPerPageOptions={props.rowsPerPageOptions}
      ActionsComponent={TablePaginationActions}
      data-qa={props['data-qa']}
    />
  );
};

export default memo(TablePagination);