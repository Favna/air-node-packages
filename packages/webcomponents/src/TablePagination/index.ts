export { default as TablePagination, TablePaginationProps, MutatedMUITablePaginationProps } from './TablePagination';