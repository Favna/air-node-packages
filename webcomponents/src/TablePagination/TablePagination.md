### Usage of the TablePagination component:

**Examples:**

```tsx
import React from 'react';
import { TablePagination } from '@rws-air/webcomponents';

   <TablePagination
          rowsPerPageOptions={tableRowsPerPage}
          rowsPerPage={props.rowsPerPage}
          page={props.page}
          count={props.rowCount}
          onChangePage={props.onChangePage}
          onChangeRowsPerPage={props.onChangeRowsPerPage}
          customClasses={css.tableTopPagination} 
    />
```

