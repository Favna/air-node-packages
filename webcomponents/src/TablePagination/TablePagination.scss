@import '../styles/vars';

.tablePagination {
  @include beautifulCenter;
  align-items: center !important;
  background-color: $light-blue !important;
  display: flex !important;
  float: left !important;
  font-size: $font-size-default !important;
  height: 2.4vw !important;
  width: 100% !important;

  &Mobile {
    height: 2.5rem !important;
  }
}

.selectIcon {
  left: 1.4rem !important;
}

.menuItem {
  font-size: $font-size-default !important;
}
