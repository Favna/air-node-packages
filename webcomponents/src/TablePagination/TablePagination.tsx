import React, { FC, ChangeEvent, ReactNode } from "react";
import MUITablePagination, { TablePaginationProps as MUITablePaginationProps } from '@material-ui/core/TablePagination';
import classnames from 'classnames';
import css from './TablePagination.scss';
import TablePaginationActions from '../TablePaginationActions/TablePaginationActions';
import useMediaQuery from '@material-ui/core/useMediaQuery';

export type TablePaginationProps = {
    labelRowsPerPage: string | ReactNode;
    labelPaginationOf: string;
    rowsPerPageOptions: number[];
    rowsPerPage: number;
    page: number;
    count: number;
    customClasses?: string | string[];
    onChangePage: (event: any, page: number) => void;
    onChangeRowsPerPage: (event: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => void;
} & MUITablePaginationProps

const TablePagination: FC<TablePaginationProps> = props => {
    const isOnMobile = useMediaQuery('(max-width: 1024px)');

    return (
        <MUITablePagination
            component='div'
            count={props.count} page={props.page}
            labelRowsPerPage={props.labelRowsPerPage}
            labelDisplayedRows={({ from, to, count }) => `${from <= 9 ? `0${from}` : from}-${to} ${props.labelPaginationOf} ${count}`}
            onChangePage={props.onChangePage} onChangeRowsPerPage={props.onChangeRowsPerPage}
            className={classnames(css.tablePagination, { [css.tablePaginationMobile]: isOnMobile }, props.customClasses)}
            classes={{ selectIcon: css.selectIcon, menuItem: css.menuItem }}
            rowsPerPage={props.rowsPerPage} rowsPerPageOptions={props.rowsPerPageOptions}
            ActionsComponent={TablePaginationActions}
        />
    );
};

export default TablePagination;
