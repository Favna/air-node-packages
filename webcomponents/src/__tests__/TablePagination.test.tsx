import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';
import TablePagination from '../TablePagination/TablePagination';

let tablePagination: ShallowWrapper;
const mockOnChangePage = jest.fn();
const mockOnchangeRowsPerPage = jest.fn();

beforeAll(() => tablePagination = shallow(
  <TablePagination
    labelrowsperpage='rows per page' labelpaginationof='of'
    rowsPerPageOptions={[5, 10]} rowsPerPage={5}
    count={20} page={0}
    onChangePage={mockOnChangePage} onChangeRowsPerPage={mockOnchangeRowsPerPage}
  />
));

test('should match snapshot', () => {
  expect(tablePagination).toMatchSnapshot();
});