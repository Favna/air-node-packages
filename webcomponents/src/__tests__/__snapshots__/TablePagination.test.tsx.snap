// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<WithStyles(ForwardRef(TablePagination))
  ActionsComponent={[Function]}
  className=""
  classes={
    Object {
      "menuItem": undefined,
      "selectIcon": undefined,
    }
  }
  component="div"
  count={20}
  labelDisplayedRows={[Function]}
  onChangePage={[MockFunction]}
  onChangeRowsPerPage={[MockFunction]}
  page={0}
  rowsPerPage={5}
  rowsPerPageOptions={
    Array [
      5,
      10,
    ]
  }
/>
`;
